java/src/main/java/scenarios/crewcreation/CharacterCreation.java

```diff
@@ -0,0 +1,250 @@
+                assignedAbilityScores.get(CHARISMA)
+        );
+    }
+
+    private static Map<String, Integer> assignAbilityScores(List<Integer> scoresRolled) {
+        int end = 0;
+        Map<String, Integer> abilityScoreMap = new LinkedHashMap<>();
+
+        while (end == 0) {
+            abilityScoreMap.put(PHYSICAL, null);
+            abilityScoreMap.put(AGILITY, null);
+            abilityScoreMap.put(BLASTER, null);
+            abilityScoreMap.put(PILOTING, null);
+            abilityScoreMap.put(REPAIR, null);
+            abilityScoreMap.put(CHARISMA, null);
+
+            List<Integer> availableScores = new ArrayList<>(scoresRolled);
+            for (Map.Entry<String, Integer> ability : abilityScoreMap.entrySet()) {
+                String abilityName = ability.getKey();
+                System.out.println("");
+                System.out.println("Which score would you like to assign to your [" + abilityName + "] ability?");
+                System.out.println("Enter the number corresponding to the value you wish to use...");
+
+                // TODO: Depending on how Lanterna is set up, we should potentially extract the following code into a
+                //  utility method for players to select a choice from an ordered list
+                DataPrinter.printAsOrderedList(availableScores);
+
+                Integer selectedScore = null;
+                while (selectedScore == null) {
+                    String invalidInputMessage = "Please enter a number from 1 to " + availableScores.size() + ".";
+
+                    try {
+                        selectedScore = Integer.parseInt(scanner.nextLine());
+                    } catch (NumberFormatException e) {
+                        System.out.println(invalidInputMessage);
+                        continue;
+                    }
+                    if (selectedScore < 1 || selectedScore > availableScores.size()) {
+                        System.out.println(invalidInputMessage);
+                        selectedScore = null;
+                    }
+                }
+
+                abilityScoreFlavorText(availableScores.get(selectedScore - 1), abilityName);
+                ability.setValue(availableScores.get(selectedScore - 1));
+                availableScores.remove(selectedScore - 1);
+            }
+
+            printAbilityScoresWithNames(abilityScoreMap);
+            System.out.println("Are these the ability scores you wish to use? (yes/no)");
+            String answer = scanner.nextLine();
+            if (answer.equalsIgnoreCase("yes") || answer.equalsIgnoreCase("y")) {
+                end = 1;
+                break;
+            } else if (answer.equalsIgnoreCase("no") || answer.equalsIgnoreCase("n")) {
+                System.out.println("Let's try that again then. From the top.");
+            } else {
+                System.out.println("Please answer with (yes/no).");
+            }
+        }
+        return abilityScoreMap;
+    }
+
```

Comment:
This method should be private, and follow the Java Naming Conventions linked elsewhere in the review. Method names should be verbs. `printAbilityScoreFlavorText` would be good. 

This method is a perfect candidate for a [Switch](https://www.baeldung.com/java-switch) statement. You can create a separate method for each ability's flavor text and call them from your switch. 
> Keep in mind that the methods to print the text for the individual scores won't be able to use a switch, since switch cases don't work with conditional statements, but solely when the value being passed into the switch matches one of the values spec'd in the cases.

Btw, the all caps ability score names here are referencing the string fields I declared at the top of the class,. Those fields are considered [Constants](https://dzone.com/articles/how-to-define-constants-in-java), they're set to a specific value and are not supposed to change. Naming convention for constants is to use all caps.